### ISSUE TITLE
[📖 Docs]: Educate the user, minimally, on the mechanics of the WebDriver protocol

### ISSUE THREAD

**Issue by nmggithub** [webdriverio/webdriverio#10766]
### Pre-check

- [X] I know I can edit the docs but prefer to file this issue

### Describe the improvement

Missing documentation

### Description of the improvement / report

While I understand QA engineers may/should not need to know too much technically about their tools, WebDriverIO is a tool that is often used in a larger chain, one that may have more technical error messages from other tools in the chain. These errors might be confusing to the engineers, so, in lieu of good error messages from external connections, some basic education on the actual protocol may help.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by christian-bromann** [comment]
Thanks for reporting!

We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by sushantsoni5392** [comment]
I can help here, but what exactly is expected? Can you summarize, please?
1. What is WebDriver protocol?
2. How does WebDriver work?

**Comment by nmggithub** [comment]
> I can help here, but what exactly is expected? Can you summarize, please?
> 
> 1. What is WebDriver protocol?
> 2. How does WebDriver work?

Are you asking me these questions? I, myself, know the answers. I was suggesting that the answers to these questions be added to the documentation.

**Comment by sushantsoni5392** [comment]
I was asking what is expected in the documentation, just these 2 questions or anything else?

**Comment by nmggithub** [comment]
> I was asking what is expected in the documentation, just these 2 questions or anything else?

I think those would be good for now. As more of a dev than a QA person, I am unaware of the level of specificity that would be best to provide QA engineers. That is why I left the details as an exercise to whoever took this up, as they would likely know better than me on that front.

**Comment by christian-bromann** [comment]
How about we add these information to [this](https://webdriver.io/docs/api/webdriver) page by editing `scripts/protocols.ts` and adding a markdown prefix for `webdriver`.

@sushantsoni5392 would you like to take a stab at it, happy to help out.

**Comment by erwinheitzman** [comment]
We actually already have this (see: https://v5.webdriver.io/docs/flowchartsoverview.html )

I saw that a file and link was missing on the website and fixed this locally but it seems like the charts are no longer rendering in the current version of the website. I tried taking the latest version of Mermaid but it didn't help sadly.

Christian didn't you help set this up back then in collaboration with a contributor? I really don't know anything about Mermaid and find it really unclear to be honest 👀

**Comment by christian-bromann** [comment]
@erwinheitzman the flow charts render fine for me: https://webdriver.io/docs/contribute/#flowcharts

**Comment by erwinheitzman** [comment]
Oh you're right, seems like there are some absolute files and paths in our repo (old references?). I wonder that if you combine those charts with the information found here https://webdriver.io/docs/automationProtocols what is still missing or unclear

**Comment by christian-bromann** [comment]
@sushantsoni5392 still interested?

**Comment by christian-bromann** [comment]
@sushantsoni5392 that is ok, thanks for letting us know.

**Comment by christian-bromann** [comment]
@nmggithub we have this document: https://webdriver.io/docs/automationProtocols#webdriver-protocol is there anything you are missing?

**Comment by nmggithub** [comment]
@christian-bromann from an initial glance it looks pretty good! Thank you for this!

**Comment by christian-bromann** [comment]
Great, closing then. Happy to extend these docs anytime.
